src/world_interface/dummy_cv_classes.h

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+namespace cam {
+// dummy CameraParams class, so the type isn't incomplete anymore
+class CameraParams {};
+} // namespace cam
+
+namespace cv {
+// same as above
+class Mat {};
+} // namespace cv
```

Comment:
I realized that this solution, while it works, isn't really necessary.

My original thinking was this:
If you included opencv headers in the same compilation unit then yeah they would cause naming conflicts. However, the point is that code should either include AR/OpenCV code _or_ this header file. This header file's only purpose is to allow the implementations of `world_interface.h` that don't use AR to compile. The `readCamera()` and co. functions aren't used by anyone in these build targets, so it doesn't matter that the `cv::Mat` and `cam::CameraParams` classes are just empty classes.

However, I just realized that since nobody calls these functions, there's no need to even define them. Therefore, we can skip this trick and just remove the function definitions from world interface implementations that don't use cameras.